Source/WebCore/platform/graphics/WidthIterator.cpp

```diff
@@ -524,7 +524,7 @@ auto WidthIterator::calculateAdditionalWidth(GlyphBuffer& glyphBuffer, GlyphBuff
             rightAdditionalWidth += m_fontCascade->letterSpacing();
 
         if (treatAsSpace && (character != tabCharacter || !m_run->allowTabs()) && (currentCharacterIndex || character == noBreakSpace) && m_fontCascade->wordSpacing())
-            rightAdditionalWidth += m_fontCascade->wordSpacing();
+            rightAdditionalWidth += m_fontCascade->wordSpacing() * m_run->effectiveZoom();
```

Comment:
Shouldn't `wordSpacing` itself resolve to the right value? Why do we need to apply an effective zoom here?
IIRC, this bug didn't happen for non-relative font-size units, right? I believe wordSpacing should be affected by the used font size, which is already taking zoom into consideration.